### ISSUE TITLE
Admin scripts don't tweak sys.path

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6208]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)| @therve reported|
|-|-|
|Trac ID|trac#6208|
|Type|defect|
|Created|2012-12-14 14:38:23Z|
|Branch|https://github.com/twisted/twisted/tree/admin-scripts-path-6208|

Scripts in bin/admin don't insert the current path in sys.path, meaning we can end up using the system install twisted instead of the current one when making a release.

<details><summary>Searchable metadata</summary>

```
trac-id__6208 6208
type__defect defect
reporter__therve therve
priority__normal normal
milestone__ 
branch__branches_admin_scripts_path_6208 branches/admin-scripts-path-6208
branch_author__therve therve
status__closed closed
resolution__fixed fixed
component__release_management release management
keywords__easy easy
time__1355495903000000 1355495903000000
changetime__1360399344000000 1360399344000000
version__None None
owner__therve therve
cc__radix
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

(In [36785]) Branching to 'admin-scripts-path-6208'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve removed owner|
|-|-|

Alright, fixed in the branch. I wondered about tests, but those scripts don't have `--version` (and I don't think they should), so I can't use the existing infrastructure for it.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_4"></a>|@tomprince set owner to @therve|
|-|-|

Looking at the tests for the other scripts, it looks like it only tests that the the checked-out version of twisted is used *or* that the installed version of twisted has the same version number as the checkout out one. It is probably not worth it to fix that.

I've manually tested this, by locally changing the version, and printing out the version at the beginning of each `main` method called by the script, which is probably enough, but could be automated.

Feel free to merge as-is, or add some tests as suggested above and resubmit for review.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set status to `closed`|
|-|-|

(In [37076]) Merge admin-scripts-path-6208

Author: therve
Reviewer: tom.prince
Fixes: [#6208](https://github.com/twisted/twisted/issues/6208)

Make sure admin scripts use the current checkout when running.
